### PR TITLE
Lvnodcdxi 17

### PR DESCRIPTION
**Pull request by MaxVoloskiy**
Created UI of Component with sliders

### CODE REVIEW

File: src/client/components/componentWithSliders/allSliders/allSliders.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  { name: 'Distortion', numOfSliders: 2 },
+  { name: 'Flenger', numOfSliders: 4 },
+  { name: 'Reverb', numOfSliders: 2 },
+  { name: 'Delay', numOfSliders: 3 },
+  { name: 'Tremble', numOfSliders: 1 }
+];
+
+const AllSliders = () => {
+  const blocks = [];
+  for (let i = 0; i < BLOCKS.length; i++) {
```
Comment (tonypntsrn) [inline src/client/components/componentWithSliders/allSliders/allSliders.js]: there is a better way to render elements in a loop in react, you can refer to this docs https://reactjs.org/docs/lists-and-keys.html
Better will be to map through your elements and return an array of jsx
If more info recruited, contact me

File: src/client/components/componentWithSliders/allSliders/blockOfSliders/blockOfSliders.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './blockOfSliders.css';
+import OneSlider from '../slider/slider';
+
+const BlockOfSliders = (props) => {
+  const sliders = [];
+  for (let i = 0; i < props.numOfSliders; i++) {
```
Comment (tonypntsrn) [inline src/client/components/componentWithSliders/allSliders/blockOfSliders/blockOfSliders.js]: Same for here, see comment above

File: src/client/components/componentWithSliders/componentWithSliders.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import './componentWithSliders.css';
+
+import SelectItem from './selectItem';
+import AllSliders from './allSliders';
+import SaveButton from './saveButton';
+import UploadButton from './uploadButton';
+import Song from './song';
+
+const ComponentWithSliders = () => {
+  return (
```
Comment (tonypntsrn) [inline src/client/components/componentWithSliders/componentWithSliders.js]: You can avoid return statement here, and simple use arrow functions return, func = () => ();

File: src/client/components/componentWithSliders/allSliders/allSliders.js (hunk 1 above)
Comment (MaxVoloskiy) [reply src/client/components/componentWithSliders/allSliders/allSliders.js]: fixed

File: src/client/components/componentWithSliders/allSliders/blockOfSliders/blockOfSliders.js (hunk 2 above)
Comment (MaxVoloskiy) [reply src/client/components/componentWithSliders/allSliders/blockOfSliders/blockOfSliders.js]: fixed

File: src/client/components/componentWithSliders/componentWithSliders.js (hunk 3 above)
Comment (MaxVoloskiy) [reply src/client/components/componentWithSliders/componentWithSliders.js]: fixed for all

File: src/client/components/componentWithSliders/allSliders/blockOfSliders/blockOfSliders.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+const renderSliders = (numOfSliders) => {
+  const sliders = [];
+  for (let i = 0; i < numOfSliders; i++) {
+    sliders.push(<OneSlider key={i}/>);
+  }
+  return sliders;
+};
+
+const BlockOfSliders = (props) => (
```
Comment (tonypntsrn) [inline src/client/components/componentWithSliders/allSliders/blockOfSliders/blockOfSliders.js]: here all needed props could be districted like this const blocksliders = ({ name, numOfSliders}) = ()

File: src/client/components/componentWithSliders/allSliders/allSliders.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  { name: 'Distortion', numOfSliders: 2 },
+  { name: 'Flenger', numOfSliders: 4 },
+  { name: 'Reverb', numOfSliders: 2 },
+  { name: 'Delay', numOfSliders: 3 },
+  { name: 'Tremble', numOfSliders: 1 }
+];
+
+const AllSliders = () => (
+  <div className="Sliders">
+    {BLOCKS.map((block, i) => {
```
Comment (tonypntsrn) [inline src/client/components/componentWithSliders/allSliders/allSliders.js]: You can avoid return statement here

File: src/client/components/componentWithSliders/allSliders/blockOfSliders/blockOfSliders.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  <div className="Sliders__block">
+    <p className="Sliders__block--title">{props.name}</p>
+    <div className="Sliders__block--sliders">
+      {renderSliders(props.numOfSliders)}
```
Comment (tonypntsrn) [inline src/client/components/componentWithSliders/allSliders/blockOfSliders/blockOfSliders.js]: 👍

File: src/client/components/componentWithSliders/allSliders/blockOfSliders/blockOfSliders.js (hunk 4 above)
Comment (MaxVoloskiy) [reply src/client/components/componentWithSliders/allSliders/blockOfSliders/blockOfSliders.js]: fixed

File: src/client/components/componentWithSliders/allSliders/allSliders.js (hunk 5 above)
Comment (MaxVoloskiy) [reply src/client/components/componentWithSliders/allSliders/allSliders.js]: fixed

**Review by tonypntsrn — APPROVED** [review]

**Pull request merged**